controllers/mongodb/shops.js

```diff
@@ -0,0 +1,108 @@
+const Product = require("../models/product");
+
+// get product list page
+exports.getProducts = (req, res, next) => {
+  Product.fetchAll()
+    .then((products) => {
+      res.render("shop/product-list", {
+        prods: products,
+        pageTitle: "All Products",
+        path: "/products",
+      });
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+};
+
+// get product detail page
+exports.getProduct = (req, res, next) => {
+  const prodId = req.params.productId;
```

Comment:
try to use - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

  `const prodId = req?.params?.productId;`